tests/backend_test.py

```diff
@@ -1235,3 +1237,43 @@ def test_ecr_gate_compilation(ibm_sherbrooke_backend: IBMQBackend) -> None:
             circ, optimisation_level
         )
         assert ibm_sherbrooke_backend.valid_circuit(compiled_circ)
+
+
+# helper function for testing
+def _get_qiskit_statevector(qc: QuantumCircuit) -> np.ndarray:
+    """Given a QuantumCircuit, use aer_simulator_statevector to compute its
+    statevector, return the vector with its endianness adjusted"""
+    back = Aer.get_backend("aer_simulator_statevector")
+    qc.save_state()
+    job = back.run(qc)
+    return np.array(job.result().data()["statevector"].reverse_qargs().data)
+
+
+# https://github.com/CQCL/pytket-qiskit/issues/99
+def test_statevector_simulator_gateset() -> None:
+    sv_backend = AerStateBackend()
+    sv_supported_gates = sv_backend.backend_info.gate_set
+    assert OpType.Reset and OpType.Measure in sv_supported_gates
+    assert OpType.Conditional in sv_supported_gates
+    # This circuit is deterministic despite the non-unitary ops.
```

Comment:
Need to be careful what we mean by "deterministic" here. Usually it means that the results of the final measurements are deterministic, but here we actually mean it has a well-defined (non-mixed) final state.